### ISSUE TITLE
geckodriver 0.15 and selenium 3.2.0 and mozilla 52 not working

### ISSUE THREAD

**Issue by akiselev87** [SeleniumHQ/selenium#3633]
I had a Eclipse + Java test framework with selenium 2.53.0 which was working with 46.0 mozilla browser.

Now I updated selenium to 3.2.0 version, mozilla to the last 52.0 (32 bit) and added geckodriver:

```
	<dependency>
    <groupId>org.seleniumhq.selenium</groupId>
    <artifactId>selenium-server</artifactId>
    <version>3.2.0</version>
    </dependency> 
	
	<dependency>
    <groupId>org.seleniumhq.selenium</groupId>
    <artifactId>selenium-java</artifactId>
    <version>3.2.0</version>
    </dependency> 
```
and
```
else if(testBrowser.equalsIgnoreCase("Mozilla")&& existingIEBrowser==null){
			//To Load Firefox driver Instance. 
			System.setProperty("webdriver.gecko.driver","geckodriver.exe");
			driver = new FirefoxDriver();
			existingmozillaBrowser=driver;
			SuiteBase.Add_Log.info("Firefox Driver Instance loaded successfully.");
			SuiteBase.extentTest.log(LogStatus.INFO, "Firefox Driver Instance loaded successfully.");
```

I got this in console: 

> INFO [main] (SeleniumWebDriver.java:26) - Test browser: Mozilla
> 1489081509368	geckodriver	INFO	Listening on 127.0.0.1:29510
> 1489081510003	mozprofile::profile	INFO	Using profile path C:\Users\akise\AppData\Local\Temp\rust_mozprofile.B36uA5Q9KvCC
> 1489081510016	geckodriver::marionette	INFO	Starting browser C:\Program Files (x86)\Mozilla Firefox\firefox.exe with args []
> 1489081510035	geckodriver::marionette	INFO	Connecting to Marionette on localhost:52883
> 1489081510332	addons.manager	DEBUG	Application has been upgraded
> 1489081510390	addons.manager	DEBUG	Loaded provider scope for resource://gre/modules/addons/XPIProvider.jsm: ["XPIProvider"]
> 1489081510392	addons.manager	DEBUG	Loaded provider scope for resource://gre/modules/LightweightThemeManager.jsm: ["LightweightThemeManager"]
> 1489081510395	addons.manager	DEBUG	Loaded provider scope for resource://gre/modules/addons/GMPProvider.jsm
> 1489081510397	addons.manager	DEBUG	Loaded provider scope for resource://gre/modules/addons/PluginProvider.jsm
> 1489081510397	addons.manager	DEBUG	Starting provider: XPIProvider
> 1489081510398	addons.xpi	DEBUG	startup
> 1489081510398	addons.xpi	INFO	SystemAddonInstallLocation directory is missing
> 1489081510399	addons.xpi	INFO	Mapping aushelper@mozilla.org to C:\Program Files (x86)\Mozilla Firefox\browser\features\aushelper@mozilla.org.xpi
> 1489081510400	addons.xpi	INFO	Mapping e10srollout@mozilla.org to C:\Program Files (x86)\Mozilla Firefox\browser\features\e10srollout@mozilla.org.xpi
> 1489081510400	addons.xpi	INFO	Mapping firefox@getpocket.com to C:\Program Files (x86)\Mozilla Firefox\browser\features\firefox@getpocket.com.xpi
> 1489081510400	addons.xpi	INFO	Mapping webcompat@mozilla.org to C:\Program Files (x86)\Mozilla Firefox\browser\features\webcompat@mozilla.org.xpi
> 1489081510401	addons.xpi	INFO	Mapping {972ce4c6-7e08-4474-a285-3208198ce6fd} to C:\Program Files (x86)\Mozilla Firefox\browser\extensions\{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi
> 1489081510402	addons.xpi	DEBUG	checkForChanges
> 1489081510402	addons.xpi	INFO	SystemAddonInstallLocation directory is missing
> 1489081510402	addons.xpi	DEBUG	Loaded add-on state from prefs: {}
> 1489081510403	addons.xpi	DEBUG	New add-on aushelper@mozilla.org in app-system-defaults
> 1489081510404	addons.xpi	DEBUG	getModTime: Recursive scan of aushelper@mozilla.org
> 1489081510405	addons.xpi	DEBUG	New add-on e10srollout@mozilla.org in app-system-defaults
> 1489081510405	addons.xpi	DEBUG	getModTime: Recursive scan of e10srollout@mozilla.org
> 1489081510406	addons.xpi	DEBUG	New add-on firefox@getpocket.com in app-system-defaults
> 1489081510406	addons.xpi	DEBUG	getModTime: Recursive scan of firefox@getpocket.com
> 1489081510407	addons.xpi	DEBUG	New add-on webcompat@mozilla.org in app-system-defaults
> 1489081510407	addons.xpi	DEBUG	getModTime: Recursive scan of webcompat@mozilla.org
> 1489081510408	addons.xpi	DEBUG	New add-on {972ce4c6-7e08-4474-a285-3208198ce6fd} in app-global
> 1489081510408	addons.xpi	DEBUG	getModTime: Recursive scan of {972ce4c6-7e08-4474-a285-3208198ce6fd}
> 1489081510409	addons.xpi	DEBUG	getInstallState changed: true, state: {"app-system-defaults":{"aushelper@mozilla.org":{"d":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\aushelper@mozilla.org.xpi","st":1489079267063},"e10srollout@mozilla.org":{"d":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\e10srollout@mozilla.org.xpi","st":1489079267062},"firefox@getpocket.com":{"d":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\firefox@getpocket.com.xpi","st":1489079267058},"webcompat@mozilla.org":{"d":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\webcompat@mozilla.org.xpi","st":1489079266981}},"app-global":{"{972ce4c6-7e08-4474-a285-3208198ce6fd}":{"d":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\extensions\\{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi","st":1489079267065}}}
> 1489081510417	addons.xpi-utils	DEBUG	Opening XPI database C:\Users\akise\AppData\Local\Temp\rust_mozprofile.B36uA5Q9KvCC\extensions.json
> 1489081510419	addons.xpi-utils	DEBUG	New add-on aushelper@mozilla.org installed in app-system-defaults
> *** Blocklist::_loadBlocklistFromFile: blocklist is disabled
> 1489081510429	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510430	addons.xpi-utils	DEBUG	New add-on e10srollout@mozilla.org installed in app-system-defaults
> 1489081510433	DeferredSave.extensions.json	DEBUG	Starting timer
> 1489081510435	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510436	addons.xpi-utils	DEBUG	New add-on firefox@getpocket.com installed in app-system-defaults
> 1489081510443	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510443	addons.xpi-utils	DEBUG	New add-on webcompat@mozilla.org installed in app-system-defaults
> 1489081510447	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510448	addons.xpi-utils	DEBUG	New add-on {972ce4c6-7e08-4474-a285-3208198ce6fd} installed in app-global
> 1489081510452	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510452	addons.manager	DEBUG	Registering startup change 'installed' for aushelper@mozilla.org
> 1489081510461	addons.xpi	DEBUG	Loading bootstrap scope from C:\Program Files (x86)\Mozilla Firefox\browser\features\aushelper@mozilla.org.xpi
> 1489081510467	addons.xpi	DEBUG	Calling bootstrap method install on aushelper@mozilla.org version 2.0
> 1489081510467	addons.xpi-utils	DEBUG	Make addon app-system-defaults:aushelper@mozilla.org visible
> 1489081510468	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510469	addons.manager	DEBUG	Registering startup change 'installed' for e10srollout@mozilla.org
> 1489081510469	addons.xpi	DEBUG	Loading bootstrap scope from C:\Program Files (x86)\Mozilla Firefox\browser\features\e10srollout@mozilla.org.xpi
> 1489081510474	addons.xpi	DEBUG	Calling bootstrap method install on e10srollout@mozilla.org version 1.9
> 1489081510476	addons.xpi-utils	DEBUG	Make addon app-system-defaults:e10srollout@mozilla.org visible
> 1489081510476	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510477	addons.manager	DEBUG	Registering startup change 'installed' for firefox@getpocket.com
> 1489081510478	addons.xpi	DEBUG	Loading bootstrap scope from C:\Program Files (x86)\Mozilla Firefox\browser\features\firefox@getpocket.com.xpi
> 1489081510483	addons.xpi	DEBUG	Calling bootstrap method install on firefox@getpocket.com version 1.0.5
> 1489081510483	addons.xpi-utils	DEBUG	Make addon app-system-defaults:firefox@getpocket.com visible
> 1489081510483	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510483	addons.manager	DEBUG	Registering startup change 'installed' for webcompat@mozilla.org
> 1489081510484	addons.xpi	DEBUG	Loading bootstrap scope from C:\Program Files (x86)\Mozilla Firefox\browser\features\webcompat@mozilla.org.xpi
> 1489081510486	addons.xpi	DEBUG	Calling bootstrap method install on webcompat@mozilla.org version 1.0
> 1489081510486	addons.xpi-utils	DEBUG	Make addon app-system-defaults:webcompat@mozilla.org visible
> 1489081510487	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510487	addons.xpi-utils	DEBUG	Make addon app-global:{972ce4c6-7e08-4474-a285-3208198ce6fd} visible
> 1489081510487	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510488	addons.xpi	DEBUG	Updating XPIState for {"id":"aushelper@mozilla.org","syncGUID":"{6c191cdb-2f89-45a4-a14d-e7a971f8121a}","location":"app-system-defaults","version":"2.0","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Application Update Service Helper","description":"Sets value(s) in the update url based on custom checks.","creator":null,"homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\aushelper@mozilla.org.xpi","installDate":1489079267063,"updateDate":1489079267063,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":8486,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"52.0","maxVersion":"52.*"}],"targetPlatforms":[],"multiprocessCompatible":true,"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false,"mpcOptedOut":false}
> 1489081510488	addons.xpi	DEBUG	getModTime: Recursive scan of aushelper@mozilla.org
> 1489081510489	addons.xpi	DEBUG	Updating XPIState for {"id":"e10srollout@mozilla.org","syncGUID":"{476ecbc2-86c6-444b-a062-23f6f377348a}","location":"app-system-defaults","version":"1.9","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Multi-process staged rollout","description":"Staged rollout of Firefox multi-process feature.","creator":null,"homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\e10srollout@mozilla.org.xpi","installDate":1489079267062,"updateDate":1489079267062,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":7181,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"52.0","maxVersion":"52.*"}],"targetPlatforms":[],"multiprocessCompatible":true,"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false,"mpcOptedOut":false}
> 1489081510489	addons.xpi	DEBUG	getModTime: Recursive scan of e10srollout@mozilla.org
> 1489081510490	addons.xpi	DEBUG	Updating XPIState for {"id":"firefox@getpocket.com","syncGUID":"{effbf822-0767-4f17-833b-1e9fff63479d}","location":"app-system-defaults","version":"1.0.5","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Pocket","description":"When you find something you want to view later, put it in Pocket.","creator":null,"homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\firefox@getpocket.com.xpi","installDate":1489079267058,"updateDate":1489079267058,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":920279,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"52.0","maxVersion":"52.*"}],"targetPlatforms":[],"multiprocessCompatible":true,"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false,"mpcOptedOut":false}
> 1489081510490	addons.xpi	DEBUG	getModTime: Recursive scan of firefox@getpocket.com
> 1489081510491	addons.xpi	DEBUG	Updating XPIState for {"id":"webcompat@mozilla.org","syncGUID":"{778a6caf-60dd-48ca-8912-5bfeccf09a73}","location":"app-system-defaults","version":"1.0","type":"extension","internalName":null,"updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Web Compat","description":"Urgent post-release fixes for web compatibility.","creator":null,"homepageURL":null},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\features\\webcompat@mozilla.org.xpi","installDate":1489079266981,"updateDate":1489079266981,"applyBackgroundUpdates":1,"bootstrap":true,"skinnable":false,"size":1454,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":false,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"52.0","maxVersion":"52.*"}],"targetPlatforms":[],"multiprocessCompatible":true,"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false,"mpcOptedOut":false}
> 1489081510491	addons.xpi	DEBUG	getModTime: Recursive scan of webcompat@mozilla.org
> 1489081510492	addons.xpi	DEBUG	Updating XPIState for {"id":"{972ce4c6-7e08-4474-a285-3208198ce6fd}","syncGUID":"{98e8b9fb-d0e1-497e-ad8e-13901523082e}","location":"app-global","version":"52.0","type":"theme","internalName":"classic/1.0","updateURL":null,"updateKey":null,"optionsURL":null,"optionsType":null,"aboutURL":null,"icons":{"32":"icon.png","48":"icon.png"},"iconURL":null,"icon64URL":null,"defaultLocale":{"name":"Default","description":"The default theme.","creator":"Mozilla","homepageURL":null,"contributors":["Mozilla Contributors"]},"visible":true,"active":true,"userDisabled":false,"appDisabled":false,"descriptor":"C:\\Program Files (x86)\\Mozilla Firefox\\browser\\extensions\\{972ce4c6-7e08-4474-a285-3208198ce6fd}.xpi","installDate":1489079267065,"updateDate":1489079267065,"applyBackgroundUpdates":1,"skinnable":true,"size":19581,"sourceURI":null,"releaseNotesURI":null,"softDisabled":false,"foreignInstall":false,"hasBinaryComponents":false,"strictCompatibility":true,"locales":[],"targetApplications":[{"id":"{ec8030f7-c20a-464f-9b0e-13a3a9e97384}","minVersion":"52.0","maxVersion":"52.0"}],"targetPlatforms":[],"seen":true,"dependencies":[],"hasEmbeddedWebExtension":false}
> 1489081510492	addons.xpi	DEBUG	getModTime: Recursive scan of {972ce4c6-7e08-4474-a285-3208198ce6fd}
> 1489081510494	DeferredSave.extensions.json	DEBUG	Save changes
> 1489081510495	addons.xpi	DEBUG	Updating database with changes to installed add-ons
> 1489081510495	addons.xpi-utils	DEBUG	Updating add-on states
> 1489081510496	addons.xpi-utils	DEBUG	Writing add-ons list
> 1489081510502	addons.xpi	DEBUG	Registering manifest for C:\Program Files (x86)\Mozilla Firefox\browser\features\aushelper@mozilla.org.xpi
> 1489081510503	addons.xpi	DEBUG	Calling bootstrap method startup on aushelper@mozilla.org version 2.0
> 1489081510505	addons.xpi	DEBUG	Registering manifest for C:\Program Files (x86)\Mozilla Firefox\browser\features\e10srollout@mozilla.org.xpi
> 1489081510506	addons.xpi	DEBUG	Calling bootstrap method startup on e10srollout@mozilla.org version 1.9
> 1489081510506	addons.xpi	DEBUG	Registering manifest for C:\Program Files (x86)\Mozilla Firefox\browser\features\firefox@getpocket.com.xpi
> 1489081510507	addons.xpi	DEBUG	Calling bootstrap method startup on firefox@getpocket.com version 1.0.5
> 1489081510509	addons.xpi	DEBUG	Registering manifest for C:\Program Files (x86)\Mozilla Firefox\browser\features\webcompat@mozilla.org.xpi
> 1489081510509	addons.xpi	DEBUG	Calling bootstrap method startup on webcompat@mozilla.org version 1.0
> 1489081510512	addons.manager	DEBUG	Registering shutdown blocker for XPIProvider
> 1489081510512	addons.manager	DEBUG	Provider finished startup: XPIProvider
> 1489081510512	addons.manager	DEBUG	Starting provider: LightweightThemeManager
> 1489081510512	addons.manager	DEBUG	Registering shutdown blocker for LightweightThemeManager
> 1489081510512	addons.manager	DEBUG	Provider finished startup: LightweightThemeManager
> 1489081510513	addons.manager	DEBUG	Starting provider: GMPProvider
> 1489081510520	addons.manager	DEBUG	Registering shutdown blocker for GMPProvider
> 1489081510521	addons.manager	DEBUG	Provider finished startup: GMPProvider
> 1489081510521	addons.manager	DEBUG	Starting provider: PluginProvider
> 1489081510521	addons.manager	DEBUG	Registering shutdown blocker for PluginProvider
> 1489081510521	addons.manager	DEBUG	Provider finished startup: PluginProvider
> 1489081510522	addons.manager	DEBUG	Completed startup sequence
> 1489081510866	Marionette	INFO	Listening on port 52883
> 1489081511201	addons.manager	DEBUG	Starting provider: <unnamed-provider>
> 1489081511201	addons.manager	DEBUG	Registering shutdown blocker for <unnamed-provider>
> 1489081511202	addons.manager	DEBUG	Provider finished startup: <unnamed-provider>
> 1489081511232	addons.manager	DEBUG	Starting provider: PreviousExperimentProvider
> 1489081511232	addons.manager	DEBUG	Registering shutdown blocker for PreviousExperimentProvider
> 1489081511233	addons.manager	DEBUG	Provider finished startup: PreviousExperimentProvider
> 1489081511236	DeferredSave.extensions.json	DEBUG	Starting write
> 1489081511517	DeferredSave.extensions.json	DEBUG	Write succeeded
> 1489081511518	addons.xpi-utils	DEBUG	XPI Database saved, setting schema version preference to 19
> FAILED CONFIGURATION: @AfterMethod reporterDataResults([TestResult name=suiteManageDictAddDict status=FAILURE method=DICT01.suiteManageDictAddDict(java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String, java.lang.String)[pri:0, instance:solution.testSuite.ManageDictionary.DICT01@4c15e7fd] output={null}])
> java.lang.NullPointerException

So, browser opens at this step: driver = new FirefoxDriver(); and fails in exact same second (@AfterMethod fails cause driver is null).

What should I do to fix this bug?

Thanks!

**Comment by AutomatedTester** [comment]
This is waiting on 3.3.1 release.

**Comment by akiselev87** [comment]
Thanks for the response. So, what is the best set of version for geckodriver, selenium and Mozilla for today?

**Comment by AutomatedTester** [comment]
there are issues with the Java client. Once that has shipped, release Firefox, v0.15 Geckodriver and the unreleased 3.3.1.

**Comment by AutomatedTester** [comment]
duplicate of #3632
